docs/design-documents/platform/pin-names/pin_names.md

```diff
@@ -0,0 +1,198 @@
+
+### Arduino Uno (Rev3) Header Pins
+
+All Arduino Uno (Rev3) form factor Mbed controller boards should define D0-D15 & A0-A5 pins as a default standard.
+
+    // Arduino Uno (Rev3) header pin connection naming  
+    // Px_xx relates to the processor pin connected to the Arduino Uno (Rev3) header pin
+    
+    D0 = Px_xx,  // UART RX / GPIO 0  
+    D1 = Px_xx,  // UART TX / GPIO 1  
+    D2 = Px_xx,  // GPIO 2  
+    D3 = Px_xx,  // TMR / PWM / GPIO 3  
+    D4 = Px_xx,  // GPIO 4  
+    D5 = Px_xx,  // TMR / PWM / GPIO 5  
+    D6 = Px_xx,  // TMR / PWM / GPIO 6  
+    D7 = Px_xx,  // GPIO 7  
+    D8 = Px_xx,  // GPIO 8  
+    D9 = Px_xx,  // TMR / PWM / GPIO 9  
+    D10 = Px_xx, // SPI CS / TMR / PWM / GPIO 10  
+    D11 = Px_xx, // SPI MOSI / TMR / PWM / GPIO 11  
+    D12 = Px_xx, // SPI MISO / GPIO 12  
+    D13 = Px_xx, // SPI SCK / GPIO 13  
+    D14 = Px_xx, // I2C SDA / GPIO 14  
+    D15 = Px_xx, // I2C SCL / GPIO 15  
+    
+    A0 = Px_xx,  // ADC 0 / GPIO 16  
+    A1 = Px_xx,  // ADC 1 / GPIO 17  
+    A2 = Px_xx,  // ADC 2 / GPIO 18  
+    A3 = Px_xx,  // ADC 3 / GPIO 19  
+    A4 = Px_xx,  // ADC 4 / GPIO 20  
+    A5 = Px_xx,  // ADC 5 / GPIO 21
+
+**I2C Definition for Arduino Uno (Rev3) Pins**
+
+All I2C definitions for the Arduino Uno (Rev3) header pins should be defined as follows:
+
+    // Arduino Uno (Rev3) header pin connection naming   
+    // Px_xx relates to the processor pin connected to the Arduino Uno (Rev3) header pin
+    
+    D14 = Px_xx, // I2C SDA / GPIO 14  
+    D15 = Px_xx, // I2C SCL / GPIO 15
+
+    // I2C signals are aliased as below  
+    I2C_SDA = D14, // I2C SDA  
+    I2C_SCL = D15, // I2C SCL
+
+**SPI definition for Arduino Uno (Rev3) Pins**  
+
+All SPI definitions for the Arduino Uno (Rev3) header pins should be defined as follows:
+
+    // Arduino Uno (Rev3) header pin connection naming  
+    // Px_xx relates to the processor pin connected to the Arduino Uno (Rev3) header pin
+    
+    D10 = Px_xx,    // SPI CS / TMR / PWM / GPIO 10  
+    D11 = Px_xx,    // SPI MOSI / TMR / PWM / GPIO 11  
+    D12 = Px_xx,    // SPI MISO / GPIO 12  
+    D13 = Px_xx,    // SPI SCK / GPIO 13  
+
+    // SPI signals are aliased as below  
+    
+    SPI_CS = D10,   // SPI CS  
+    SPI_MOSI = D11, // SPI MOSI  
+    SPI_MISO = D12, // SPI MISO  
+    SPI_SCK = D13,  // SPI SCK
+
+### Custom/Vendor Header Pins
+
+**Custom/Vendor Connector I2C Definition**
+
+All I2C definitions for a new connector should be defined as follows:
+
+    // ConnectorName is the name of the specific connector you are implementing signals for here  
+    // Px_xx relates to the processor pin connected to the custom/vendor connector pin  
+    
+    ConnectorName_I2C0_SCL  = Px_xx,  // Connector I2C0 SCL  
```

Comment:
Connector pin is connected to a MCU pin,
so name should be more something like 
ConnectorName_ConnectorPin = Px_xx,

Then you could define some alias:
ConnectorName_I2C_SCL = ConnectorName_ConnectorPin

Reply:
My reading of that was that this wasn't a function alias, it was the actual signal name for this connector.

Reply:
Agree with @kjbracey-arm on this. However, vendors could choose to define connector pins as "ConnectorName_I2C_SCL = ConnectorName_ConnectorPin". Aliasing is okay as long as pin names are defined, for the sake of portability. For example, if multiple boards use the same connector, then ConnectorName_ConnectorPin should be defined, whether as an alias or processor pin.

Reply:
I'd like us to evaluate the possibility to use namespacing to have clean definition and usage.
What do you think about the following?
- For Arduino connectors: `I2C i2c(ArduinoUno::I2C_SDA , ArduinoUno::I2C_SCL);`
- For MCU pins: `I2C i2c(ADI::MCU::PB_9 , ADI::MCU::PB_8);`

Reply:
Using the namespaces 👍  c++1x scoped enum or similar would be useful?

Reply:
We need to review implications on using namespaces as it's a C++ specific feature.
The current HAL is based on C - we may want to keep it for compatibility reasons and just use explicit defines for each of the connector standards.